lib/redis/RedisCache.ts

```diff
@@ -74,19 +73,31 @@ export class RedisCache<T> implements GroupedCache<T>, Cache<T>, Loader<T> {
   }
 
   async clear(): Promise<void> {
-    const cacheItems = await this.executeWithTimeout(this.redis.keys(this.resolveCachePattern()))
-    const chunks = chunk(cacheItems, 1000)
-    for (let chunk of chunks) {
-      await this.executeWithTimeout(this.redis.del(chunk))
-    }
+    const pattern = this.resolveCachePattern()
+    let cursor = '0'
+    do {
+      const scanResults = await this.executeWithTimeout(this.redis.scan(cursor, 'MATCH', pattern))
+
+      cursor = scanResults[0]
+      if (scanResults[1].length === 0) {
+        return
+      }
+      await this.executeWithTimeout(this.redis.del(scanResults[1]))
+    } while (cursor !== '0')
   }
 
   async deleteGroup(group: string) {
-    const itemsInGroup = await this.executeWithTimeout(this.redis.keys(this.resolveKeyGroupPattern(group)))
-    const chunks = chunk(itemsInGroup, 1000)
-    for (let chunk of chunks) {
-      await this.executeWithTimeout(this.redis.del(chunk))
-    }
+    const pattern = this.resolveKeyGroupPattern(group)
+    let cursor = '0'
+    do {
+      const scanResults = await this.executeWithTimeout(this.redis.scan(cursor, 'MATCH', pattern))
+
+      cursor = scanResults[0]
+      if (scanResults[1].length === 0) {
+        return
```

Comment:
MATCH is evaluated right before sending results to client.
Therefore there can be empty sets of data while performing and returning scanned result

Reply:
![image](https://user-images.githubusercontent.com/3844817/221855260-de71df54-8589-4b13-94ca-a5265eb0d144.png)